faunadb/client_test.go

```diff
@@ -2382,6 +2382,112 @@ func (s *ClientTestSuite) TestEchoQuery() {
 	s.Require().True(firstSeen > 0 && s.client.GetLastTxnTime() >= firstSeen)
 }
 
+func (s *ClientTestSuite) TestPaginateAccessProviders() {
+	var obj f.ObjectV
+	var val f.Value
+	var arr f.Arr
+	var b bool
+
+	name := f.RandomStartingWith("name_")
+	issuer := f.RandomStartingWith("https://xxxx.auth0.com")
+	jwksUri := f.RandomStartingWith("https://xxxx.auth0.com/")
+	//allowed_roles := f.Arr{}
+	//allowed_collections := f.Arr{}
+
+	query := f.CreateAccessProvider(f.Obj{
+		"name":     name,
+		"issuer":   issuer,
+		"jwks_uri": jwksUri,
+	})
+	s.adminQueryAndDecode(query, &obj)
+	obj.At(refField).Get(&ref)
+
+	s.adminQueryAndDecode(f.Paginate(f.AccessProviders()), &val)
+	val.At(dataField).Get(&arr)
+
+	s.Require().True(len(arr) != 0)
+
+	s.adminQueryAndDecode(f.GT(f.Count(f.AccessProviders()), 0), &b)
+	s.Require().True(b)
+
+}
+func (s *ClientTestSuite) TestCreateAndReadAccessProvider() {
+	var provider1, provider2, provider3 f.ObjectV
+	var arr f.Arr
+	var ref f.RefV
+	var val f.Value
+	var b bool
+
+	name := f.RandomStartingWith("name_")
+	issuer := f.RandomStartingWith("https://no.issuer.fauna.com/")
+	jwksUri := f.RandomStartingWith("https://xxxx.auth0.com/")
+	//allowed_roles := f.Arr{}
+	//allowed_collections := f.Arr{}
+
+	s.adminQueryAndDecode(f.CreateAccessProvider(f.Obj{
+		"name":     name,
+		"issuer":   issuer,
+		"jwks_uri": jwksUri,
+	}), &provider1)
+	provider1.At(refField).Get(&ref)
+
+	s.Require().Equal(f.StringV(name), provider1["name"])
```

Comment:
I'd recommending using a struct to decode and assert the values, e.g.: https://github.com/fauna/faunadb-go/blob/b675cddbc09dcb76c15cb55b52b65d6eab13534a/faunadb/client_test.go#L261.